fix_: chats and message history loading after login takes too much time

qfrank opened the pull request:
The status community has around 60 chats. Each chat costs `initChatFirstMessageTimestamp` around 120 ms in my ios simulator, so there is around an 8-second delay for `InitFilters`. The major cause is db io consumption. To reduce this, I combined all the chat parameters into one db query and made a simple cache for community queries. Also, I call save community only once for each community. Now the delay for `InitFilters` is reduced to around 800 ms. `initChatFirstMessageTimestamp` took around 251 ms

fix mobile [issue](https://github.com/status-im/status-mobile/issues/21358)

qfrank committed:
fix_: chats and message history loading after login takes too much time

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (60)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | a0de2a0a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/1/) | 2024-10-10 01:11:40 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/1//consoleText) |
| :heavy_check_mark: | a0de2a0a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/1/) | 2024-10-10 01:12:48 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/1//consoleText) |
| :heavy_check_mark: | a0de2a0a | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/1/) | 2024-10-10 01:13:22 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241010-010828-a0de2a-pr5932.zip) |
| :heavy_check_mark: | a0de2a0a | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/1/) | 2024-10-10 01:13:52 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241010-010823-a0de2a-pr5932.aar) |
| :heavy_check_mark: | a0de2a0a | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/1/) | 2024-10-10 01:14:07 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241010-010824-a0de2a-pr5932.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 49a1d90e | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/2/) | 2024-10-10 01:17:14 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/2//consoleText) |
| :heavy_check_mark: | 49a1d90e | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/2/) | 2024-10-10 01:17:26 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241010-011452-49a1d9-pr5932.aar) |
| :heavy_check_mark: | 49a1d90e | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/2/) | 2024-10-10 01:17:57 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241010-011457-49a1d9-pr5932.zip) |
| :heavy_check_mark: | 49a1d90e | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/2/) | 2024-10-10 01:18:42 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241010-011454-49a1d9-pr5932.zip) |
| :heavy_check_mark: | 49a1d90e | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/2/) | 2024-10-10 01:20:07 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | b7c061ea | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/3/) | 2024-10-10 06:40:36 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241010-063823-b7c061-pr5932.aar) |
| :heavy_multiplication_x: | b7c061ea | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/3/) | 2024-10-10 06:40:37 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/3//consoleText) |
| :heavy_check_mark: | b7c061ea | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/3/) | 2024-10-10 06:41:16 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241010-063827-b7c061-pr5932.zip) |
| :heavy_check_mark: | b7c061ea | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/3/) | 2024-10-10 06:42:26 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241010-063824-b7c061-pr5932.zip) |
| :heavy_check_mark: | b7c061ea | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/3/) | 2024-10-10 06:43:39 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0d595b2d | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/4/) | 2024-10-10 08:53:05 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241010-085058-0d595b-pr5932.aar) |
| :heavy_check_mark: | 0d595b2d | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/4/) | 2024-10-10 08:53:45 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241010-085103-0d595b-pr5932.zip) |
| :heavy_check_mark: | 0d595b2d | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/4/) | 2024-10-10 08:55:53 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241010-085059-0d595b-pr5932.zip) |
| :heavy_check_mark: | 0d595b2d | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/4/) | 2024-10-10 08:56:55 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/4//consoleText) |
| :heavy_check_mark: | 0d595b2d | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/4/) | 2024-10-10 09:23:21 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3cd2dcf9 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/5/) | 2024-10-10 08:54:58 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241010-085319-3cd2dc-pr5932.aar) |
| :heavy_check_mark: | 3cd2dcf9 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/5/) | 2024-10-10 08:56:18 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241010-085400-3cd2dc-pr5932.zip) |
| :heavy_check_mark: | 3cd2dcf9 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/5/) | 2024-10-10 08:59:37 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241010-085605-3cd2dc-pr5932.zip) |
| :heavy_check_mark: | 3cd2dcf9 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/5/) | 2024-10-10 09:02:13 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/5//consoleText) |
| :heavy_multiplication_x: | 3cd2dcf9 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/5/) | 2024-10-10 09:28:56 | ~5 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/5//consoleText) |
| :heavy_check_mark: | 3cd2dcf9 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/6/) | 2024-10-10 10:08:30 | ~29 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 93033e8f | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/6/) | 2024-10-11 02:23:43 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241011-022056-93033e-pr5932.zip) |
| :heavy_check_mark: | 93033e8f | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/6/) | 2024-10-11 02:24:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241011-022053-93033e-pr5932.zip) |
| :heavy_check_mark: | 93033e8f | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/6/) | 2024-10-11 02:26:10 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241011-022052-93033e-pr5932.aar) |
| :heavy_check_mark: | 93033e8f | [#6](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/6/) | 2024-10-11 02:27:05 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/6//consoleText) |
| :heavy_check_mark: | 93033e8f | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/7/) | 2024-10-11 02:52:53 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 4a510a05 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/7/) | 2024-10-11 02:34:09 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241011-023153-4a510a-pr5932.aar) |
| :heavy_check_mark: | 4a510a05 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/7/) | 2024-10-11 02:34:39 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241011-023156-4a510a-pr5932.zip) |
| :heavy_check_mark: | 4a510a05 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/7/) | 2024-10-11 02:35:44 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241011-023154-4a510a-pr5932.zip) |
| :heavy_check_mark: | 4a510a05 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/7/) | 2024-10-11 02:37:11 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 055b7717 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/8/) | 2024-10-11 02:48:56 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241011-024635-055b77-pr5932.aar) |
| :heavy_check_mark: | 055b7717 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/8/) | 2024-10-11 02:49:22 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241011-024639-055b77-pr5932.zip) |
| :heavy_check_mark: | 055b7717 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/8/) | 2024-10-11 02:50:15 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241011-024637-055b77-pr5932.zip) |
| :heavy_check_mark: | 055b7717 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/8/) | 2024-10-11 02:51:50 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/8//consoleText) |
| :heavy_check_mark: | 055b7717 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/8/) | 2024-10-11 03:26:17 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 66ad879e | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/9/) | 2024-10-11 02:56:32 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241011-025342-66ad87-pr5932.aar) |
| :heavy_check_mark: | 66ad879e | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/9/) | 2024-10-11 02:56:55 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241011-025346-66ad87-pr5932.zip) |
| :heavy_check_mark: | 66ad879e | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/9/) | 2024-10-11 02:57:56 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241011-025343-66ad87-pr5932.zip) |
| :heavy_check_mark: | 66ad879e | [#9](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/9/) | 2024-10-11 02:59:11 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/9//consoleText) |
| :heavy_check_mark: | 66ad879e | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/9/) | 2024-10-11 03:59:05 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c5ba2175 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/10/) | 2024-10-11 11:05:28 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241011-110308-c5ba21-pr5932.aar) |
| :heavy_check_mark: | c5ba2175 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/10/) | 2024-10-11 11:05:51 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241011-110313-c5ba21-pr5932.zip) |
| :heavy_check_mark: | c5ba2175 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/10/) | 2024-10-11 11:07:05 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241011-110309-c5ba21-pr5932.zip) |
| :heavy_check_mark: | c5ba2175 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/10/) | 2024-10-11 11:08:48 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/10//consoleText) |
| :heavy_check_mark: | c5ba2175 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/10/) | 2024-10-11 11:35:45 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8045454b | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/11/) | 2024-10-14 06:33:02 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241014-062959-804545-pr5932.aar) |
| :heavy_check_mark: | 8045454b | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/11/) | 2024-10-14 06:33:14 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241014-063005-804545-pr5932.zip) |
| :heavy_check_mark: | 8045454b | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/11/) | 2024-10-14 06:34:39 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241014-063000-804545-pr5932.zip) |
| :heavy_check_mark: | 8045454b | [#11](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/11/) | 2024-10-14 06:35:35 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/11//consoleText) |
| :heavy_check_mark: | 8045454b | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/11/) | 2024-10-14 07:02:52 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | eb54942c | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/12/) | 2024-10-14 08:26:16 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241014-082422-eb5494-pr5932.aar) |
| :heavy_check_mark: | eb54942c | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/12/) | 2024-10-14 08:27:04 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241014-082426-eb5494-pr5932.zip) |
| :heavy_check_mark: | eb54942c | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/12/) | 2024-10-14 08:28:13 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241014-082423-eb5494-pr5932.zip) |
| :heavy_check_mark: | eb54942c | [#12](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/12/) | 2024-10-14 08:29:51 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/12//consoleText) |
| :heavy_check_mark: | eb54942c | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/12/) | 2024-10-14 08:56:31 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/12//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 4a3ddfa9 | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/13/) | 2024-10-15 06:56:20 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241015-065354-4a3ddf-pr5932.aar) |
| :heavy_check_mark: | 4a3ddfa9 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/13/) | 2024-10-15 06:56:44 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241015-065359-4a3ddf-pr5932.zip) |
| :heavy_check_mark: | 4a3ddfa9 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/13/) | 2024-10-15 06:57:56 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241015-065355-4a3ddf-pr5932.zip) |
| :heavy_check_mark: | 4a3ddfa9 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/13/) | 2024-10-15 06:59:23 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/13//consoleText) |
| :heavy_check_mark: | 4a3ddfa9 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/13/) | 2024-10-15 07:26:15 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | d6a9c398 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5932/14/) | 2024-10-16 09:27:43 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241016-092510-d6a9c3-pr5932.aar) |
| :heavy_check_mark: | d6a9c398 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5932/14/) | 2024-10-16 09:27:55 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241016-092513-d6a9c3-pr5932.zip) |
| :heavy_check_mark: | d6a9c398 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5932/14/) | 2024-10-16 09:29:38 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241016-092511-d6a9c3-pr5932.zip) |
| :heavy_check_mark: | d6a9c398 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/14/) | 2024-10-16 09:30:38 | ~5 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5932/14//consoleText) |
| :heavy_check_mark: | d6a9c398 | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/14/) | 2024-10-16 09:57:36 | ~32 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5932/14//consoleText) |

seanstrom commented on protocol/messenger.go:
```diff
@@ -2043,24 +2044,56 @@ func (m *Messenger) Mailservers() ([]string, error) {
-func (m *Messenger) initChatFirstMessageTimestamp(chat *Chat) error {
-	if !chat.CommunityChat() || chat.FirstMessageTimestamp != FirstMessageTimestampUndefined {
-		return nil
+func (m *Messenger) initChatFirstMessageTimestamp(chats []*Chat) {
+	var communityChats []*Chat
```
Maybe we could rename this function to: `initChatsFirstMessageTimestamp`

seanstrom reviewed (approved):
The changes look good, though I see there's some Codecov warnings for some of the new code. Is it possible to add some tests for functions that interact with the db?

qfrank commented:
> The changes look good, though I see there's some Codecov warnings for some of the new code. Is it possible to add some tests for functions that interact with the db?

Sure, I'll add it later thank you!

jrainville commented on protocol/messenger.go:
```diff
@@ -2043,24 +2044,60 @@ func (m *Messenger) Mailservers() ([]string, error) {
+		m.logger.Warn("failed to get oldest message timestamp", zap.Error(err))
+		return
 	}
+	communityCache := make(map[string]*communities.Community)
```
We already have the same type of cache done above called `communityInfo`. Could we move this code to the main function to reuse the cache, or pass a ref to the cache, or even move the community cache to a `Messenger` prop?

I think passing a ref is probably the simplest and cleanest

qfrank replied:
fixed

jrainville commented on protocol/messenger.go:
```diff
@@ -2043,24 +2044,60 @@ func (m *Messenger) Mailservers() ([]string, error) {
+		if err != nil {
+			m.logger.Warn("failed to save community", zap.Error(err), zap.String("communityID", community.IDString()))
+		}
+		m.communitiesManager.Publish(&communities.Subscription{Community: community})
```
I don't think it makes sense to publish here, unless you actively checked if something changed. This way, we'll just be spamming the network on every all open.

Either you should check the community really changed and then publish, or just leave the publish to the normal sending loop.

If you go for the first option, you should also use `saveAndPublish` instead. It does the saving and the publishing while also checking that you are control node.

qfrank replied:
fixed

jrainville reviewed (changes requested):
Very nice work. The speed up seems impressive.

I have some suggestions to make sure we reuse the code as best as possible and also to not publish without reason. The community description is the biggest offender in terms of using bandwidth so we need to be careful with publishing it.

qfrank committed:
chore_: split to small functions to writing unit test easily

jrainville commented on protocol/messenger.go:
```diff
@@ -2043,24 +2044,84 @@ func (m *Messenger) Mailservers() ([]string, error) {
-func (m *Messenger) initChatFirstMessageTimestamp(chat *Chat) error {
-	if !chat.CommunityChat() || chat.FirstMessageTimestamp != FirstMessageTimestampUndefined {
-		return nil
+func (m *Messenger) initChatsFirstMessageTimestamp(communityCache map[string]*communities.Community, chats []*Chat) {
```
```suggestion
func (m *Messenger) initChatsFirstMessageTimestamp(communityCache *map[string]*communities.Community, chats []*Chat) {
```
this way, you're passing the pointer/reference and you can change the original map and also it's lighter since it's not a new value

qfrank replied:
are you sure we need the extra pointer?
```go
type communityInfo struct {
	ID string
}

func TestPassMap(t *testing.T) {
	communityInfoMap := make(map[string]*communityInfo)
	communityInfoMap["1"] = &communityInfo{}
	updateCommunityInfo(communityInfoMap)
	require.Equal(t, "666", communityInfoMap["2"].ID)
}

func updateCommunityInfo(communityInfoMap map[string]*communityInfo) {
	communityInfoMap["2"] = &communityInfo{ID: "666"}
}
```

jrainville replied:
Ah, since the `communityInfo` is a pointer, it still works! All good then

jrainville reviewed (commented):
Looks good I think the only thing left would be to use a pointer, otherwise the changes you make to the cache aren't actually shared with the original cache

qfrank committed:
test_: add test

ilmotta commented on protocol/message_persistence.go:
```diff
@@ -1305,26 +1305,50 @@ func (db sqlitePersistence) MessageByChatIDs(chatIDs []string, currCursor string
+		}
+		result[chatID] = whisperTimestamp
+	}
+
```
A general recommendation:

> The error from rows.Err() could be the result of a variety of errors in the rows.Next() loop. The loop might exit for some reason other than finishing the loop normally, so you always need to check whether the loop terminated normally or not. An abnormal termination automatically calls rows.Close(), although it’s harmless to call it multiple times.
>
> -- http://go-database-sql.org/errors.html

Also here https://go.dev/doc/database/querying#multiple_rows_errors

```go
if err = rows.Err(); err != nil {
	return nil, err
}
```

qfrank replied:
fixed

ilmotta commented on protocol/message_persistence.go:
```diff
@@ -1305,26 +1305,50 @@ func (db sqlitePersistence) MessageByChatIDs(chatIDs []string, currCursor string
+		return nil, nil
 	}
+
+	args := make([]interface{}, len(chatIDs))
```
`[]interface{}` -> `[]any`

qfrank replied:
I need tell AI to use `any` over `interface{}` 😅

<img width="447" alt="image" src="https://github.com/user-attachments/assets/8faf8be3-1510-4656-8d41-c539797a942d">

emm.. the rules looks good 😊

<img width="644" alt="image" src="https://github.com/user-attachments/assets/ade51547-8d0b-421e-907b-16deb360a75f">

qfrank replied:
fixed

ilmotta replied:
Yep 😆 chatgpt refuses to use the alias without guidance.

ilmotta commented on protocol/message_persistence.go:
```diff
@@ -1305,26 +1305,50 @@ func (db sqlitePersistence) MessageByChatIDs(chatIDs []string, currCursor string
+			SELECT
+				m1.local_chat_id,
+				m1.whisper_timestamp,
+				ROW_NUMBER() OVER (PARTITION BY m1.local_chat_id ORDER BY substr('0000000000000000000000000000000000000000000000000000000000000000' || m1.clock_value, -64, 64) || m1.id ASC) AS rn
```
I believe this is the first query in status-go using `partition by`. Not only for that, but I wonder if this query performs well when there are lots of rows in `user_messages`.

By any chance, did you run a query plan analysis on this query? One idea is to generate tons of rows (1k, 10k, 100k, and 1M) from a unit test and grab the test sqlite file to analyse.

qfrank replied:
Nice question, I used follow python script to generated 1M records into `user_messages`:
```python
import sqlite3
import random
import string
import time
from datetime import datetime, timedelta

# Connect to the database
conn = sqlite3.connect('your_database.db')
cursor = conn.cursor()

# Function to generate a random string
def random_string(length):
    return ''.join(random.choices(string.ascii_letters + string.digits, k=length))

# Function to generate a random date within the last year
def random_date():
    return int((datetime.now() - timedelta(days=random.randint(0, 365))).timestamp())

# Generate 1 million records
num_records = 1000000
batch_size = 1000

start_time = time.time()

for i in range(0, num_records, batch_size):
    values = []
    for j in range(batch_size):
        if i + j >= num_records:
            break
        
        values.append((
            random_string(32),  # id
            random.randint(1600000000, int(time.time())),  # whisper_timestamp
            random_string(10),  # source
            random_string(100),  # text
            random.randint(0, 5),  # content_type
            random_string(15),  # username
            random_date(),  # timestamp
            random_string(32),  # chat_id
            random_string(32),  # local_chat_id
            random.choice([0, 1]),  # hide
            random_string(32) if random.random() < 0.1 else None,  # response_to
            random.randint(0, 5),  # message_type
            random.randint(1, 1000000),  # clock_value
            random.choice([0, 1]),  # seen
            random.choice(['sent', 'delivered', 'read']),  # outgoing_status
            random_string(50).encode(),  # parsed_text
            random.choice([0, 1]),  # rtl
            random.randint(1, 10),  # line_count
            random.choice([0, 1]),  # mentioned
            random_date() if random.random() < 0.1 else None,  # edited_at
            random.choice([0, 1]),  # deleted
            random.randint(0, 3),  # contact_request_state
            random_string(20) if random.random() < 0.1 else "",  # discord_message_id
            random.choice([0, 1]),  # deleted_for_me
            random.randint(0, 2),  # contact_verification_status
            random_string(32) if random.random() < 0.1 else None,  # album_id
            random.choice([0, 1]),  # replied
            random.randint(100, 1000) if random.random() < 0.2 else None,  # image_width
            random.randint(100, 1000) if random.random() < 0.2 else None,  # image_height
            random_string(32) if random.random() < 0.05 else None,  # deleted_by
            random.randint(1, 10) if random.random() < 0.1 else 0,  # album_images_count
        ))

    cursor.executemany('''
    INSERT INTO user_messages (
        id, whisper_timestamp, source, text, content_type, username, timestamp, chat_id, local_chat_id,
        hide, response_to, message_type, clock_value, seen, outgoing_status, parsed_text, rtl, line_count,
        mentioned, edited_at, deleted, contact_request_state, discord_message_id, deleted_for_me,
        contact_verification_status, album_id, replied, image_width, image_height, deleted_by, album_images_count
    ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
    ''', values)

    if (i + batch_size) % 100000 == 0:
        conn.commit()
        print(f"Inserted {i + batch_size} records")

conn.commit()
cursor.close()
conn.close()

end_time = time.time()
print(f"Inserted {num_records} records in {end_time - start_time:.2f} seconds")
```

Then tried following the query and it ended in 211 ms. looks good? 🤔

```sql
SELECT
    m1.local_chat_id,
    m1.whisper_timestamp
FROM (
         SELECT
             m1.local_chat_id,
             m1.whisper_timestamp,
             ROW_NUMBER() OVER (PARTITION BY m1.local_chat_id ORDER BY substr('0000000000000000000000000000000000000000000000000000000000000000' || m1.clock_value, -64, 64) || m1.id ASC) AS rn
         FROM user_messages m1
         WHERE m1.local_chat_id IN (
                                    'ngO4zPugiB3FEl89SWVRx6Iz36Z62J3q',
                                    'zpgd5Dx7exAuBnynngvmQmw8ljD8a90T',
                                    'UkFHdjB537zBT0Aloxsiqy98EC7Uyjfp',
                                    'GuRwYMcswAns8fJ11BtXahvBEg5ctoaT',
                                    '3jOElCI2uZCObP6xKWGBikrh0ZyXlZow',
                                    'aqwiwJIpJzPp7k7IkDNSAvFGnIWkRUZy',
                                    'ob3zKgnxWj5gH4whPvM6u9PUBWLG48Ai',
                                    'VR0yUJuxAXNoTegg8mAkBIMPe4ZuaF0c',
                                    'QhGx83VdQlfKA7ViWdG8pUBwZYO93Ztr',
                                    'aUIJP1eL58jqwP0J3z2h5njXJxETf3gA',
                                    '3gO0YB89LLCWXsVQHvXZHCgopZiTdKmX',
                                    '3ukti2ncPqyNHhIFX62Y2ztakrLIC9hL',
                                    'vw3pvJj1uuh1pjrYUGnBCvuxHh5vcKai',
                                    'pcij9DK3CQxjxIsRvnSezSJBSs5xNyta',
                                    '8AdaW0NRSXkzicqQee321HpQA6OyeMRL',
                                    'oqC7UEokLZSLQ8B9WuBQkQF6PCMimeTM',
                                    '2ovRmn0vIgHxvXh2bsKtOmRycK38Ck3N',
                                    'FwjwQyDLtqNcGRlCcZHbVCegqDcCL0Y3',
                                    'kvTDvfYL0ouusLooQ05ImjU27z0Qo2nd',
                                    'aNVAiZMfJ8LIGzQfRGbxK8oIRdL0f7IM',
                                    'sjX0d6moeuV7fQzUGpUAuA0Sk4lZVzji',
                                    '9ncyXL6COmw2vAf5mDzarfljou6D0wiX',
                                    'VPIuSTj8HJE0r5sZyvtPS5WxLvuF2g57',
                                    'L8XXRA9eneGIAZ8TGwCclsiFZNpyJMNw',
                                    'aXUbyxmrKp8WWidlskyKagbhlAYqCKh0',
                                    'cUXJDEgEg06BJkB2azlZuOYmdRQuFsnu',
                                    'buW1mGc3eH37gzsvJbov16zzluJjdn9M',
                                    'I7SYYWEXXaCk3JwnSYoJejCt4Vrn5WNX',
                                    'Vek9vID9knpErwMoaDRd0B3yV3AADzA1',
                                    'pd0kK8Vgjbf3GI2WdGvYwDNIICLaI4XF',
                                    'TM2m18xF212BgJNzhhZwTg1EOgoWBoy4',
                                    '9jrhtPOoZlHUrU9czXeAp2W8JZ06pI1y',
                                    'M2EOMuTb5zfQUrP09RJOiAOXedmnpzua',
                                    'kW7DqCDILHhIIgu9JrHFQO3rU8CJBFZS',
                                    'j6wLUdEb27NCDFp5OseKhM3URuXC5L7o',
                                    '6ixotqEF0slrIfFA3NPVVcTFV4L8zxg2',
                                    'NmtzmH6NtqisjSbcq9Etp9QgOW5TsaK3',
                                    '8k91kKj7deIQOWkxsFYXAmbf66UXwVIQ',
                                    'NLuDjJM7jNzSeErOzWHSdR2afx0AV65Z',
                                    'Vb7RZBvRhyVnJsJTj6RUYy3PJj0QOI69',
                                    '8tZdXC2N0akgmfrJ3ggsXzY0nUgJXz5z',
                                    'ItrAZq6eCDbgO7Ud58RVcoObIf78bIGe',
                                    'ZnUS7SFdQ3xp5BfKzEZz5VhyYELmB4yf',
                                    'iwQrgOuevZi5c3nUKhNMbyQAynlNd7pU',
                                    'QrJbHIiNqzNstmgGlzX1BjDE9hLBItvC',
                                    'G1qk0OgEMdd9hTDS8gzOvc6Htwf3Dc1Y',
                                    'zQewDmdfFfeQYHPjImcFVdtBMzpTMxJc',
                                    'D0oeKfyEL4ZFoYhRT7Lqj5EGYGku2Mtv',
                                    '3i1pGIRo48FxhKHq0Hs5t67opCPJTCxV',
                                    'syy2MT6mdAKLkUQbApvSCR4hjKdX8NiV',
                                    'CYHJdWTcFZrCMXA54SyQhjNrMP92XWjl',
                                    '7nasHxbDZh1vYYubsm0ijKQ8fSgra3nG',
                                    'TwyhsaJiPkKzW3QwZWq1DIf4Y6GLiOSR',
                                    '3ehJUDUBtvJc3gzXE2R8hNj6DQzDgnkA',
                                    '7dVA7kT3v69R6A699KK9c8xLwQcgeJ1g',
                                    'lfyJzTo4k2xw0aktIAN14039oRJVhwbJ',
                                    'syDHKMJZdDqm2zigNSZqwZ48T6icq3wW',
                                    'm1RHt8XSeGwtEDGKpPmRKSn753Ar4q3q',
                                    '82mFf9AuL4bhTcSuuWoZFf9UQRSdKxsY',
                                    'mCpmN2j1cfNYHWNOuRae05uf5T0JBZql',
                                    'OojIacDrUVlImvVIdYaWDr6ICAEW4SUQ',
                                    'VPo0yNq1qm882EDKtB9bsz2QoNSMcxca',
                                    '4z8RE3eoP4pr7QI246wWtf0Q87skFcD4',
                                    'Kfawz7FuF3sKu9g1Pr9lSGHe3XSWB4OC',
                                    '4FS2ra1X4aX8wR3TsxWbdcAPFOtOSUcA',
                                    '4Iqt0c0CkInTcuXc2guDMqqa5hVZUVsA',
                                    'GcKdEsLHn8M4XHhQmEJTPnY0Xxt3jc2K',
                                    '2RRCxgri7vpI5WVpV0uZzFIq4KaZUpct',
                                    '3dJ96CTdfRusBzufUGuUWxFlWluNFgLl',
                                    'JmoosPmVfocscLSGX4UwNXYbNpzE1SuE',
                                    'S4zLWWbk2SS6a0yt2HnRrruSpmP1YQtw',
                                    '6rd5qlMA20drnC1fmV4ZJTAliiV29JEq',
                                    'IqGgKV50am9eM5yTGzgUphzuscH7DIPu',
                                    'kfWZciWxFV5rJli2zB2yqU8E80XyxC9O',
                                    '9QZO1NqYIP9kD3dlAykaxUNmZQjekE2a',
                                    'KCQyoyaO11fI8NSJiFNjROHeenFuZvrZ',
                                    'Y6y01PpjLmkhZg85anTxnXTv76ZVIXbu',
                                    '5lZ0Q86Hx6pkOnGThO3u5FiNQDvoeYRl',
                                    'fpF5iklMGGWTYEoukGN3suWHKQZJIIn4',
                                    'RQ0DL2vMWWvdS64D4h6pu7SIRU6PbHgN',
                                    'Jwti0xidYYD4BJ0ihb1opYKvmjz1Wudr',
                                    'Pm7mUCjPtfZpnxBnjPlSKIM2er6g6bb6',
                                    'AvW7Y4vyD10WlhMDdVOpaeeHaODdc2KV',
                                    'CW6XbhpOWCGRuXbcI3ax9QyYHecRP794',
                                    'r5CxNP0X4JhC3NdmYyQ9fv1rEK8BxPQU',
                                    '3ydbZMY4rbdVY3WHpHnqPNqZEaS0mTHa',
                                    'pNMsPsFbg9kTnqNsRetbAv4PZOYZSl9c',
                                    'uMd5JCTQOHM5qApfCVbn4Er1AUgk4jl3',
                                    'Ol9MICToav0ExKlvSMLqnaPP0noOPXno',
                                    'Zb1ToBb6yMx67ZmQZ5Gm0ezpBv8Pg3CH',
                                    'D2lD8Kv1ckawj2Cjp1mxcq9DhfZqUEFu',
                                    'Gv6hCdeBiuI6T5pdVoMkk1PdPqdNtd4s',
                                    'cWzuJ8XVZn313uva4UYywegP3aY2AkPH',
                                    '2cjGhSqG0P9H3xxIcf4yGbmuJvzIMM9M',
                                    'Gc9QF9tqfyESdkWmexYUpo9BUoDcPw30',
                                    'cQ1MLtHPgf4DRhHBh3s7nu58YFIIPHLZ',
                                    'TaDaSMtOKT9JuF9m8GlBMSpx8KbxBrRp',
                                    'GGVRmivfiWOyXLIRpz3fAM7KyyHWbKJA',
                                    '4jBz0qFCMoq5MHeBXoxiMNsB3RIsNwbf',
                                    '6JSc75WjowqdrEf2MFIm1jXxlOefkAyQ',
                                    'tAEqVRjVkLnkIEHi9681JA3l254EnPPL',
                                    'zLIfq4pvE8gsAdtBQdZwGv8b2KF7sjdZ',
                                    'lI2ILExDsksZKHyeMG2tLhaxrPYKUgPM',
                                    '3rdV72YMAT8Vel9gWd9uDqIY6Q7bMz6p',
                                    'K2JYatoRZA1C1J0KI2Z9RPNn8aPQsp2V',
                                    'bYXhMOUGsMiT7afhP6WPdUz5iMKhGhWG',
                                    'PqDWYM9lSXigx4nUlewTejwBqlhECS3c',
                                    '5Nds3yQFSBzNlXpRHopFAZL9hLuRRbNe',
                                    'shsGYJQnDMDPKIa5kZFPSqoFc4fmdMbx',
                                    'qaWYpkjUqdJAYU8OnBwkzHPT8agE0ZlC',
                                    'R2ez9875FndahzohdsEpvEAOVVdMeBrn',
                                    'i0bPdlJK414zoXDHnAFEdJGcDIwqw7cE',
                                    'bL0vZOxhFybyXI79fHFWvRM8ztUrTwSi',
                                    'Ub6s67H1KDy13UtMRR2KueI6LjS6MTi9',
                                    'RgGyZSCNfOJnEgD7hhNw5H6DSatMOO9u',
                                    'uBuWiUdNuZ2K4fGRJntAzk422o7axz9E',
                                    'NJGEO0lerQqd0ThKCZ8cNgxU26gDCr7V',
                                    'xXduO9NDvfBgO4WSfKIUMYIf9xpXFYA7',
                                    'ycdYK9F4srM4YmXa7LapKgeEPYqpObaR',
                                    'dwoKdOBQLuSBN6tH4tl0ZTatceXSB3UB',
                                    'fJdSAc4ilUWCaiXCjzgJAy79fvOXjb4H'
             )
     ) m1
WHERE m1.rn = 1;
```

<img width="842" alt="image" src="https://github.com/user-attachments/assets/a59b700a-4d15-40b0-9342-4190a7fc54cd">

we had the index(idx_search_by_local_chat_id_sort_on_cursor) already, so I think the performance should be ok.
<img width="1245" alt="image" src="https://github.com/user-attachments/assets/a3655ab4-1f03-4305-b77d-ea6886b13e44"> @ilmotta

ilmotta replied:
Lovely script ;) Seems pretty fast, although it's also running in your powerful dev machine.

> we had the index(idx_search_by_local_chat_id_sort_on_cursor) already, so I think the performance should be ok.

If you didn't see anything suspicious in the output of the query plan then it's great!

qfrank replied:
There did exist suspicion in the output of the query plan! Previous result looks good because I ignored generating massive records which contain the same local_chat_id! If I add another 1M record with the same local_chat_id, sql execution will take > 20s , after trying different sqls, I think I find the best sql:
```sql
SELECT
    m1.local_chat_id,
    m1.whisper_timestamp,
    min(substr('0000000000000000000000000000000000000000000000000000000000000000' || m1.clock_value, -64, 64) || m1.id)
FROM user_messages m1
WHERE m1.local_chat_id IN (
                           'ngO4zPugiB3FEl89SWVRx6Iz36Z62J3q',
                           ...
    )
group by m1.local_chat_id
```

it will take around 1s, and the query plan will be pretty simple now 👍
<img width="1009" alt="image" src="https://github.com/user-attachments/assets/3adf4272-d768-44e2-a97e-4dc55328adc5">

But I doubt if it will be enough for real user as time fly

ilmotta replied:
Query plan looks straightforward indeed. Nice!

ilmotta reviewed (commented):
Very nice work @qfrank!

qfrank committed:
chore_: improve OldestMessageWhisperTimestampByChatIDs function

- Use 'any' type instead of 'interface{}' for args slice
- Add error check after rows iteration

qfrank committed:
chore_: optimize OldestMessageWhisperTimestampByChatIDs query

This commit simplifies and optimizes the SQL query in the OldestMessageWhisperTimestampByChatIDs function. The changes include:

1. Removing the subquery and ROW_NUMBER() function
2. Using MIN() and GROUP BY instead of the previous approach
3. Directly selecting the required columns in a single query

These changes should improve the performance of the function, especially for large datasets, while maintaining the same functionality.
